File: boat-bay-frontend/src/app/models/boat-lint-report.ts
import { BoatSpec } from "./boat-spec";
import { BoatLintRule } from "./boat-lint-rule";

interface Range {
  start: number;
  end: number;
}

interface Pointer {
  matchingProperty: string;
}

export interface BoatViolation {
  rule: BoatLintRule;
  description: string;
  severity: string;
  lines: Range;
  pointer: Pointer;
}

export interface BoatLintReport {
  id: number;
  name: string;
  grade: string;
  passed: boolean;
  lintedOn: Date;
  version: string;
  openApi?: string;
  spec: BoatSpec;
  violations: BoatViolation[];
}

File: boat-bay-server/src/backbase/java/com/backbase/oss/boat/bay/web/views/dashboard/models/BoatLintReport.java
package com.backbase.oss.boat.bay.web.views.dashboard.models;

import java.time.ZonedDateTime;
import java.util.ArrayList;
import java.util.List;
import lombok.Data;

@Data
public class BoatLintReport {

    private Long id;

    private BoatSpec spec;

    private String name;
    private String grade;
    private boolean passed;
    private ZonedDateTime lintedOn;
    private String version;

    private String openApi;
    private List<BoatViolation> violations = new ArrayList<>();

    public boolean hasViolations() {
        return !violations.isEmpty();
    }

}